dask_expr/_collection.py

```diff
@@ -644,10 +644,7 @@ def shift(self, periods=1, freq=None, axis=0):
             raise TypeError("periods must be an integer")
 
         axis = _validate_axis(axis)
-        if axis == 1:
-            raise NotImplementedError("shift on axis 1 not supported yet")
-
-        return new_collection(Shift(self.expr, periods, freq))
+        return new_collection(Shift(self.expr, periods, freq, axis))
```

Comment:
Please create a different expression class that handles axis=1

Reply:
done, maybe you might like the `ShiftAxis0/1` naming though? https://github.com/dask-contrib/dask-expr/pull/473/commits/677594793981756e669f049253b74b29f74d4efb